chemclipse/plugins/org.eclipse.chemclipse.xxd.model/src/org/eclipse/chemclipse/xxd/model/filter/peaks/IntegratedAreaFilter.java

```diff
@@ -63,34 +63,60 @@ public boolean acceptsIPeaks(Collection<? extends IPeak> items) {
 
 		for(X peak : read) {
 			applySelectedOptions(peak.getIntegratedArea(), configuration, listener, peak);
-			checkForKeepOption(listener, configuration, peak);
 			subMonitor.worked(1);
 		}
 	}
 
 	private static <X extends IPeak> void applySelectedOptions(double peakValue, IntegratedAreaFilterSettings configuration, CRUDListener<X, IPeakModel> listener, X peak) {
 
+		boolean keepFlag = false;
+		if(configuration.getFilterTreatmentOption()==ValueFilterTreatmentOption.KEEP_PEAK) {
+			keepFlag = true;
+		}
```

Comment:
@kerner1000 Is it easier to understand? The flag is used to virtually invert the deletion of the peaks.